orangecontrib/text/corpus_to_network.py

```diff
@@ -0,0 +1,292 @@
+                                          self.word2ind[tokens[i]])] += 1
+                    else:
+                        self.word_matrix[(self.word2ind[tokens[i]],
+                                          self.word2ind[tokens[j]])] += 1
+                for j in range(i + 1, right + 1):
+                    if ((self.word2ind[tokens[j]], self.word2ind[tokens[i]])
+                            in self.word_matrix):
+                        self.word_matrix[(self.word2ind[tokens[j]],
+                                          self.word2ind[tokens[i]])] += 1
+                    else:
+                        self.word_matrix[(self.word2ind[tokens[i]],
+                                          self.word2ind[tokens[j]])] += 1
+                stop = self._set_progress_or_stop(next(ticks))
+                if stop:
+                    raise RuntimeError("Interruption requested.")
+
+    def _generate_word_network(self):
+        stop = self._set_progress_or_stop(90.0)
+        if stop:
+            raise RuntimeError("Interruption requested.")
+        th = self.word_threshold
+        data = np.array([v / 2 for v in self.word_matrix.values() if (v / 2) >= th],
+                        dtype=np.float64)
+        row_ind = np.array([k[0] for k, v in self.word_matrix.items() if (v / 2) >= th],
+                           dtype=np.float64)
+        col_ind = np.array([k[1] for k, v in self.word_matrix.items() if (v / 2) >= th],
+                           dtype=np.float64)
+        s = len(self.word_freqs)
+        edges = csr_matrix((data, (row_ind, col_ind)), shape=(s, s))
+        ind2word = {v:k for k, v in self.word2ind.items()}
+        words = np.array([ind2word[ind] for ind in range(s)])
+        freqs = np.array([self.word_freqs[ind2word[ind]] for ind in range(s)],
+                         dtype=np.float64)
+        network = Network(nodes=words,
+                          edges=edges,
+                          name='Word Network')
+        self.word_network = network.subgraph(self.mask)
+        domain = Domain([ContinuousVariable('word_frequency')],
+                        None,
+                        [StringVariable('word')])
+        self.word_items = Table(domain,
+                                freqs.reshape((-1, 1))[self.mask],
+                                None,
+                                words.reshape((-1, 1))[self.mask])
+        self._set_progress_or_stop(100.0)
+
+    def get_current_items(self, document_nodes):
+        """Returns table containing data about nodes of 
+        currently saved network.
+
+        Parameters
+        ----------
+        document_nodes : bool
+            If True, returns data for document network, otherwise for word
+            network.
+
+        Returns
+        -------
+        Table
+            Table with additional data about nodes.
+        """
+        return self.document_items if document_nodes else self.word_items
+
+    def _set_progress_or_stop(self, progress):
```

Comment:
This function should go to the widget and be passed to the script as a callback. This way script is more flexible for future use and changes